src/Data/Text/Utf8.hs

```diff
@@ -211,6 +281,32 @@ decode4 :: CodeUnit -> CodeUnit -> CodeUnit -> CodeUnit -> Int
 decode4 cu0 cu1 cu2 cu3 =
   (fromIntegral cu0 .&. 0x7) `shiftL` 18 .|. (fromIntegral cu1 .&. 0x3f) `shiftL` 12 .|. (fromIntegral cu2 .&. 0x3f) `shiftL` 6 .|. (fromIntegral cu3 .&. 0x3f)
 
+indexCodePoint :: Text -> CodeUnitIndex -> (Int, CodePoint)
+indexCodePoint (Text u8data off _len) (CodeUnitIndex index) =
```

Comment:
I don't see how this would establish additional safety over the unsafe function it calls, so maybe it should also have an unsafe in the name?